src/main/java/com/cdancy/bitbucket/rest/features/BranchApi.java

```diff
@@ -114,4 +119,35 @@ Branch getDefault(@PathParam("project") String project,
     @GET
     BranchModel model(@PathParam("project") String project,
                       @PathParam("repo") String repo);
+
+    @Named("branch:get-BranchPermission")
+    @Documentation({"https://developer.atlassian.com/static/rest/bitbucket-server/4.14.1/bitbucket-ref-restriction-rest.html#idm45354011023456"})
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/branch-permissions/2.0/projects/{project}/repos/{repo}/restrictions")
+    @Fallback(BitbucketFallbacks.BranchPermissionPageOnError.class)
+    @GET
+    BranchPermissionPage getBranchPermission(@PathParam("project") String project,
+                                             @PathParam("repo") String repo,
+                                             @Nullable @QueryParam("start") Integer start,
+                                             @Nullable @QueryParam("limit") Integer limit);
+
+    @Named("branch:update-BranchPermission")
+    @Documentation({"https://developer.atlassian.com/static/rest/bitbucket-server/4.14.1/bitbucket-ref-restriction-rest.html#idm45354011023456"})
+    @Path("/branch-permissions/2.0/projects/{project}/repos/{repo}/restrictions")
+    @Produces("application/vnd.atl.bitbucket.bulk+json")
```

Comment:
Yes because Bitbucket need for update. If, I don't put it, bitbucket return this error : "Can not deserialize instance of com.atlassian.stash.internal.repository.ref.restriction.rest.RestRestrictionRequest out of START_ARRAY token\n at [Source: com.atlassian.stash.internal.web.util.web.CountingServletInputStream@68f4c6d2; line: 1, column: 1]""